tfsdk/state.go

```diff
@@ -101,6 +102,22 @@ func (s *State) RemoveResource(ctx context.Context) {
 	s.Raw = tftypes.NewValue(s.Schema.Type().TerraformType(ctx), nil)
 }
 
+// TransformDefaults traverses the schema, identifies any attributes which are null, and
+// if the attribute has a default value specified by the `Default` field on the attribute
+// then the default value is assigned.
+func (s *State) TransformDefaults(ctx context.Context, configRaw tftypes.Value) diag.Diagnostics {
```

Comment:
Ah, boo. I think in the RFC I had put pseudo-code that seemed to imply we should create something like this for simplicity of the code example. That's my bad as we should avoid creating anything in the exported API _here_ before merge since that this could confuse consumers about its purpose or intended uses.

We can talk through the `PlanResourceChange` RPC code changes to directly work with `fwschema.Data` types instead of working directly with a method on the exported `tfsdk.State`.

Reply:
Have refactored `PlanResourceChange` to work directly on `fwschema.Data` types.